### ISSUE TITLE
Move webmaker-metrics to mofosecure infrastructure

### ISSUE THREAD

**Issue by adamlofting** [MozillaFoundation/plan#249]
Blocks: Add retention and quarterly goals to dashboard #237 

I've assigned this to you @jdotpz but some of this is for me to do.

I think the steps required are:
- [x] Move the repo into the mozilla org https://github.com/mozilla/webmaker-metrics
- [x] Setup hosting in mofosecure infrastructure (as opposed to heroku)
- [x] Setup a subdomain for the app (suggest: https://metrics.webmaker.org )
  - [x] Needs to be `https`
- [x] Setup deployment from the repo to hosting (staging would be nice to have but not essential)
- [x] Update app to work with new hosting rather than heroku style config (need to change .env?)
- [ ] Need an alternative to Heroku's scheduler which runs a script at set times each day to crunch the latest day's worth of data. I can either code this into the app, or if there's something existing we can use that's great.
- [x] Setup / finalize mofointegration in mofosecure environment
- [x] Setup a DB alias for this
- [x] Adam needs an admin account for the new DB, and IP / VPN linked access
- [ ] Adam to create a db_user for the app with read only access to selected views in mofointegration
- [x] Get the webmaker_metrics views into the new mofointegration
- [x] Document views in the repo

**Comment by adamlofting** [comment]
Hey @jdotpz - does this feel like something we can get done in this heartbeat? (i.e by Friday)

**Comment by jdotpz** [comment]
It's potentially possible, but not that super likely.  I've got a few other fish I'm frying for this heart beat which are falling a bit behind.

**Comment by adamlofting** [comment]
Cool, I'll flag this on the planning ticket to manage expectations.

**Comment by jdotpz** [comment]
I've subbed the request to add metrics to our SSL san.

**Comment by jdotpz** [comment]
We can ghetto this if we really need it fast, even though I'm not a big fan of that route.
What that would look like:
1) The "app" would be on a standalone server, with no puppetization or automation or jenkins.  An update would require ssh'ing in, git pulling, npm installing, updating cron, etc.
2) I add an IP on the existing controller on moforegular so I can talk to the db on this side.

@simonwex your call.

**Comment by jdotpz** [comment]
I've been working on this over the weekend when I had internet, and it seems it is going to be more work to try to point to our existing puppet infra using cross-account IAM roles for S3 mofo-techops/creds/* and the like than to just setup legit infra.  

I'm working on getting a puppet server going now.

**Comment by jdotpz** [comment]
OK, puppet server is up at securegeppetto.mofoprod.net.  I'm going to create a freshly baked repo for puppet, especially for mofosecure apps.  

I created the build scripts and cleverly named them csolnew because nobody in their right mind would click on that these days.  https://jenkins.mofoprod.net/job/csolnew/3/console
The initial tag v0.0.01 is pushed up and built, currently residing in our mofoprod repo.  I'll likely move this over, but first thing is first.

**Comment by jdotpz** [comment]
https://github.com/mozilla/mofos_system_configurations and https://github.com/mozilla/mofos_systems_scripts are started, working on cleanup next.

**Comment by jdotpz** [comment]
This is the install I used for geppetto secure.

http://pastebin.mozilla.org/8500323

**Comment by jdotpz** [comment]
https://securejenkins.mofoprod.net/ is up serving from the new geppetto/jenkins combo box.  Only https will work for now.  I'll be adding jobs once I get repos cleaned up.
